clients/src/main/java/org/apache/kafka/clients/producer/internals/ProducerBatch.java

```diff
@@ -181,11 +182,11 @@ void expirationDone() {
     }
 
     int attempts() {
-        return attempts;
+        return attempts.get();
     }
 
     void reenqueued(long now) {
-        attempts++;
```

Comment:
This was only ever done from the `Sender` thread, so I think it was OK as it was. Clearer with the `AtomicInteger` and probably OK since we'd hope `reenqueued` is not a bottleneck.